error_example_test.go

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package fx_test
+
+import (
+	"errors"
+	"fmt"
+
+	"go.uber.org/fx"
+)
+
+func ExampleError() {
+	// A function that panics is both provided and invoked,
+	// but neither of these options will be called in the
+	// presence of an fx.Error.
+	panicFn := func() { panic("should not be called") }
+
+	app := fx.New(
+		fx.Provide(panicFn),
+		// An fx.Error option was registered with the application,
+		// so none of the Provide or Invoke options will be run.
+		fx.Error(errors.New("failed to build module")),
+		fx.Invoke(panicFn),
+	)
+
+	fmt.Println(app.Err())
+
+	// Output:
+	// failed to build module
+}
```

Comment:
Discussed offline - this example feels a little odd because there's no reason for the owner of `func main` to copy it. It might be better to write from the perspective of a module author. One possible example is starting an HTTP server and reading the port from the environment - if `$PORT` isn't set, we return an error instead of a `Provide`.